### PR TITLE
better error message for non-unique columns

### PR DESCRIPTION
**Pull request by jrenner**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `git diff upstream/master | flake8 --diff`
- [ ] whatsnew entry

I ran into this error with my code and the error message didn't help me figure out I was dealing with duplicate columns that couldn't be handled. I think this is more helpful

### CODE REVIEW

File: pandas/formats/style.py [hunk 1]
```diff
@@ -137,8 +137,10 @@ def __init__(self, data, precision=None, table_styles=None, uuid=None,
             raise TypeError("``data`` must be a Series or DataFrame")
         if data.ndim == 1:
             data = data.to_frame()
-        if not data.index.is_unique or not data.columns.is_unique:
+        if not data.index.is_unique:
```
Comment (sinhrks) [inline pandas/formats/style.py]: can u add tests to check errors using ``tm.assertRaisesRegexp``.

**Comment by jreback** [comment]
can you rebase / update

**Comment by jreback** [comment]
can you rebase / update?

**Comment by jreback** [comment]
needs tests. pls reopen / comment if you can update.

**Pull request closed without merging**